addons/web_editor/static/tests/test_wysiwyg_collaboration.js

```diff
@@ -401,6 +401,49 @@ QUnit.module('web_editor', {
                 assert.equal(await peers.p2.getValue(), `[]<p>ab</p>`, 'p2 should have the same document as p1');
                 assert.equal(await peers.p3.getValue(), `<p>a[]</p>`, 'p3 should not have the document changed because it has not focused');
 
+                removePeers(peers);
+            });
+            QUnit.test('Focused client should be able to delete selected text without issue', async (assert) => {
+                assert.expect();
+                const pool = await createPeers(['p1', 'p2']);
+                const peers = pool.peers;
+
+                await peers.p1.startEditor();
+                await peers.p2.startEditor();
+
+                await peers.p1.focus();
+                await peers.p2.focus();
+
+                await peers.p1.wysiwyg.odooEditor.execCommand('insert', 'r6f6afwef6wef1wef');
+
+                await peers.p1.openDataChannel(peers.p2);
+
+                // select last 5 characters of last insertion
+                const selection = peers.p1.document.getSelection();
+                const pElement = peers.p1.wysiwyg.odooEditor.editable.querySelector('p');
+                const lastChild = pElement.lastChild;
+                const lastChildLength = lastChild.textContent.length;
+                const range = new Range();
+                range.setStart(lastChild, lastChildLength - 5);
+                range.setEnd(lastChild, lastChildLength);
+                selection.removeAllRanges();
+                selection.addRange(range);
+
+                // select end of last insertion
+                const selection2 = peers.p2.document.getSelection();
+                const pElement2 = peers.p2.wysiwyg.odooEditor.editable.querySelector('p');
+                const range2 = new Range();
+                range2.setStart(pElement2, 2);
+                range2.setEnd(pElement2, 2);
```

Comment:
Why `2`? If `r6f6afwef6wef1wef` is a single `TextNode` inside of P then the position at the end of it is `[P, 1]`, not `[P, 2]` or did I miss something? Or it can also be expressed as `[the_text_node, the_text_node.length]`. Maybe try both see if you can reproduce the traceback with one of those?

Reply:
Because by default, there is a textNode with the value "a" when selecting in the test
I rewrote the test using lastchild to be similar with the last block of code 
The issue still remains, the test is always green

Reply:
I don't know exactly why the bug does not occur in your test, but I had a look at the issue and it appears the traceback happens because `getDeepestPosition` is given a node that is in fact a lone TextNode detached from the DOM (only reachable through its reference in the mutation records), which is why `closestBlock(node)` returns `null` and the code crashes afterwards. Since your fix is to make `getDeepestPosition` resilient to be given such an input, I think the easiest way to write a test for it would be to add a test for `getDeepestPosition` in `utils.test.js` for this particular edge case. Thanks!